mod-source-record-storage-server/src/main/java/org/folio/dao/RecordDaoImpl.java

```diff
@@ -420,39 +520,33 @@ public Flowable<Row> streamMarcRecordIds(ParseLeaderResult parseLeaderResult, Pa
     }
     /* Building a count query */
     SelectJoinStep countQuery = DSL.select(countDistinct(RECORDS_LB.EXTERNAL_ID)).from(RECORDS_LB);
-    appendJoin(countQuery, parseLeaderResult, parseFieldsResult);
+    appendJoin(countQuery, parseLeaderResult);
     appendWhere(countQuery, parseLeaderResult, parseFieldsResult, searchParameters);
     /* Join both in one query */
-    String sql = DSL.select().from(searchQuery).rightJoin(countQuery).on(DSL.trueCondition()).getSQL(ParamType.INLINED);
-
+    String sql = "";
+    if (parseFieldsResult.isEnabled()) {
+      sql = DSL.with(commonTableExpression).select().from(searchQuery).rightJoin(countQuery).on(DSL.trueCondition()).getSQL(ParamType.INLINED);
+    } else {
+      sql = select().from(searchQuery).rightJoin(countQuery).on(DSL.trueCondition()).getSQL(ParamType.INLINED);
+    }
+    String finalSql = sql;
```

Comment:
@Aliaksandr-Fedasiuk Maybe it would be better to log finalSql? In debug level, for example.

Reply:
thanks for the idea!